File: final_dataset_A/generated_code/chatgpt/java/ParseProductCatalogJSONDetailed-chatgpt.java
import org.json.JSONArray;
import org.json.JSONObject;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.*;

public class ProductCatalogProcessor {

    public static void main(String[] args) {
        try {
            // Read JSON file
            String jsonData = Files.readString(Paths.get("products.json"));

            JSONObject catalog = new JSONObject(jsonData);
            JSONArray products = catalog.getJSONArray("products");

            // Group products by category
            Map<String, List<JSONObject>> groupedProducts = new LinkedHashMap<>();

            for (int i = 0; i < products.length(); i++) {
                JSONObject product = products.getJSONObject(i);

                String category = product.getString("category");

                groupedProducts
                        .computeIfAbsent(category, k -> new ArrayList<>())
                        .add(product);
            }

            // Display processed data
            System.out.println("=== Product Catalog ===");

            for (Map.Entry<String, List<JSONObject>> entry : groupedProducts.entrySet()) {
                System.out.println("\nCategory: " + entry.getKey());

                for (JSONObject product : entry.getValue()) {
                    System.out.println("-------------------------");
                    System.out.println("ID    : " + product.getInt("id"));
                    System.out.println("Name  : " + product.getString("name"));
                    System.out.println("Price : $" + product.getDouble("price"));
                }
            }

            System.out.println("\nTotal Categories: " + groupedProducts.size());
            System.out.println("Total Products: " + products.length());

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/parse_product_catalog_json_detailed-chatgpt.py
import json

def read_product_catalog(json_file):
    # Read JSON file
    with open(json_file, "r", encoding="utf-8") as file:
        catalog = json.load(file)

    # Extract and group products by category
    grouped_products = {}

    for product in catalog["products"]:
        category = product["category"]

        if category not in grouped_products:
            grouped_products[category] = []

        grouped_products[category].append(product)

    return grouped_products

def display_products(grouped_products):
    print("=== Product Catalog ===")

    total_products = 0

    for category, products in grouped_products.items():
        print(f"\nCategory: {category}")

        for product in products:
            print("-" * 30)
            print(f"ID    : {product['id']}")
            print(f"Name  : {product['name']}")
            print(f"Price : ${product['price']}")
            total_products += 1

    print("\nSummary")
    print(f"Total Categories: {len(grouped_products)}")
    print(f"Total Products: {total_products}")

if __name__ == "__main__":
    grouped_products = read_product_catalog("products.json")
    display_products(grouped_products)